drivers/pio/sunxi-pio.c

```diff
@@ -98,21 +93,14 @@ sunxi_pio_probe(struct device *dev)
 
 	/* Disable IO for all pins. */
 	for (size_t port = 0; port < MAX_PORTS; ++port) {
-		for (size_t pin = 0; pin < MAX_PINS_PER_PORT; pin += 8) {
-			mmio_write32(dev->regs +
-			             CONFIG_REG(SUNXI_PIO_PIN(port, pin)),
-			             0x77777777);
-		}
+		for (size_t pin = 0; pin < MAX_PINS_PER_PORT; pin += 8)
+			mmio_write32(dev->regs + CONFIG_REG(pin), 0x77777777);
```

Comment:
This is wrong. Here, `pin` is an offset *within* the port. Determining the register requires knowing both the port and the pin, like the original code did.

Reply:
I have changed this back to  _your_ original code. I believe we can find a better way to implement this, but that is something that should be iterated on in a later point in time.

Reply:
Er, yes, by "original code" I meant the `-` side of the diff, not the first version